src/network/wait-online/manager.c

```diff
@@ -92,7 +92,7 @@ static int manager_link_is_online(Manager *m, Link *l, const LinkOperationalStat
                                             l->state);
 
         const LinkOperationalStateRange *range;
-        FOREACH_ARGUMENT(range, state_range, &m->required_operstate, &l->required_operstate)
+        FOREACH_ARGUMENT(range, state_range, &m->required_operstate, &l->required_operstate, POINTER_MAX)
```

Comment:
Hmm, this would result in invalid memory access since it's actually passed to `operation_state_is_valid`? Maybe move the assertion below into the loop?

Reply:
Ah, indeed.